### PR TITLE
improve: optimize executor by flipping a few comparisons

### PR DESCRIPTION
**Pull request by mrice32**

### CODE REVIEW

File: src/utils/FillUtils.ts [hunk 1]
```diff
@@ -77,11 +77,11 @@ export function isFirstFillForDeposit(fill: Fill): boolean {
...
 export function filledSameDeposit(fillA: Fill, fillB: Fill): boolean {
   return (
-    fillA.amount.eq(fillB.amount) &&
+    fillA.depositId === fillB.depositId &&
     fillA.originChainId === fillB.originChainId &&
+    fillA.amount.eq(fillB.amount) &&
     fillA.destinationChainId === fillB.destinationChainId &&
     fillA.relayerFeePct.eq(fillB.relayerFeePct) &&
-    fillA.depositId === fillB.depositId &&
     fillA.recipient === fillB.recipient &&
```
Comment (nicholaspai) [inline src/utils/FillUtils.ts:85]: We could probably move recipient above originChainId? Addresses are less likely to be repeated than chain ID?

File: src/utils/FillUtils.ts [hunk 2]
```diff
@@ -106,7 +106,7 @@ export function getFillDataForSlowFillFromPreviousRootBundle(
 ) {
   // Find the first fill chronologically for matched deposit for the input fill.
   const firstFillForSameDeposit = allValidFills.find(
-    (_fill: FillWithBlock) => isFirstFillForDeposit(_fill as Fill) && filledSameDeposit(_fill, fill)
+    (_fill: FillWithBlock) => filledSameDeposit(_fill, fill) && isFirstFillForDeposit(_fill as Fill)
```
Comment (nicholaspai) [inline src/utils/FillUtils.ts:109]: Found this same one, +1000

**Review by nicholaspai — APPROVED** [review]
LGTM nice little finds

File: src/utils/FillUtils.ts (hunk 1 above)
Comment (mrice32) [reply src/utils/FillUtils.ts:85]: I think we should put these two first:

1. I think numerical comparisons are the cheapest by far, so it's way better to try to short circuit before getting to the more expensive comparisons.
2. depositId + originChainId basically filters down on the deposit. Deposits usually have 1 fill, and occasionally have 2, so these two alone should be almost a perfect filter with no string or BN comparisons.

**Comment by mrice32** [comment]
Going to go ahead and merge, just so the dataworker starts operating better in prod. We can tweak these later!

**Pull request merged**